app/src/main/java/nu/yona/app/YonaApplication.java

```diff
@@ -100,12 +102,27 @@ public void onCreate()
 		super.onCreate();
 		mContext = this;
 		eventChangeManager = new EventChangeManager();
-		sharedAppPreferences = eventChangeManager.getSharedPreference().getUserPreferences();
-		sharedUserPreferences = eventChangeManager.getSharedPreference().getAppPreferences();
+		validateAndInitializePreferences();
 		sharedAppDataState = eventChangeManager.getDataState();
 		Foreground.init(this);
 	}
 
+	private void validateAndInitializePreferences()
+	{
+		sharedAppPreferences = eventChangeManager.getSharedPreference().getAppPreferences();
+		sharedUserPreferences = eventChangeManager.getSharedPreference().getUserPreferences();
+		if (sharedUserPreferences.getString(AppConstant.SERVER_URL, null) != null)
+		{  // Need to exchange the preferences as corrective action, as a couple of previous builds had wrong(Exchanged Preferences) initializations.
```

Comment:
It's unclear to me what this code block does. The comment doesn't help me (don't understand what "Exchanged Preferences") and I don't understand why the code below does something useful. When I'm not mistaken, it moves the preferences to a temp location and back.

Please enlighten me

Reply:
`sharedAppPreferences = eventChangeManager.getSharedPreference().getUserPreferences();
		sharedUserPreferences = eventChangeManager.getSharedPreference().getAppPreferences();`

There were the statements earlier. App Preferences were copied into userPreferences variable and viceversa. This was causing the app to get logged out as it does not find logged in user.
Current code will find if preferences are wrongly initialized or not. If yes, they will be exchanged otherwise will continue. 
Exchanging is needed if user comes from previous build versions where wrong code is used.